src/client/java/teammates/client/scripts/DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat.java

```diff
@@ -0,0 +1,166 @@
+            }
+        }
+
+        println("\n################# Analysis Results #################");
+        println("Total number of comments: " + loopHelper.getCount());
+        println("Total number of documents: " + numberOfDocuments + "\n");
+
+        println("Number of documents already in new date format: " + numberOfUnaffectedDocuments);
+        println("Number of documents in unrecognizable date format: "
+                + (numberOfDocuments - commentsToFix.size() - numberOfUnaffectedDocuments) + "\n");
+
+        println("Number of documents in old date format: " + commentsToFix.size());
+        println("####################################################\n");
+
+        return commentsToFix;
+    }
+
+    private Document getDocument(FeedbackResponseCommentAttributes comment) {
+        return index.get(comment.getId().toString());
+    }
+
+    private String extractSampleDateString(Document document) {
+        String frcaJson = document.getOnlyField(Const.SearchDocumentField.FEEDBACK_RESPONSE_COMMENT_ATTRIBUTE).getText();
+        return jsonParser.parse(frcaJson).getAsJsonObject().getAsJsonPrimitive("createdAt").getAsString();
+    }
+
+    private synchronized boolean isInDateFormat(String dateString, DateFormat dateFormat) {
+        try {
+            dateFormat.parse(dateString);
+            return true;
+        } catch (ParseException e) {
+            return false;
+        }
+    }
+
+    private void fixDocuments(List<FeedbackResponseCommentAttributes> commentsToFix) {
+        println("Running data migration for " + commentsToFix.size() + " documents in old date format...");
+        println("Preview: " + isPreview);
+
+        List<Document> documentsToUpdate = new ArrayList<>();
+
+        LoopHelper loopHelper = new LoopHelper(batchSize, "documents processed.");
+
+        for (FeedbackResponseCommentAttributes commentToFix : commentsToFix) {
+            loopHelper.recordLoop();
+            queueDocumentUpdate(commentToFix, documentsToUpdate);
+
+            if (documentsToUpdate.size() == batchSize) {
+                updateAndClearDocuments(documentsToUpdate);
+            }
+        }
+        updateAndClearDocuments(documentsToUpdate);
+
+        println("\nComplete! If there are any errors shown above, please rerun this script.");
+    }
+
+    private void queueDocumentUpdate(FeedbackResponseCommentAttributes comment, List<Document> documentsToUpdate) {
+        documentsToUpdate.add(new FeedbackResponseCommentSearchDocument(comment).build());
+    }
+
+    private void updateAndClearDocuments(List<Document> documentsToUpdate) {
+        if (documentsToUpdate.isEmpty()) {
+            return;
+        }
+
+        println("Batch updating " + documentsToUpdate.size() + " documents...");
+
+        if (!isPreview) {
+            SearchManager.putDocuments(Const.SearchIndex.FEEDBACK_RESPONSE_COMMENT, documentsToUpdate);
+        }
```

Comment:
SearchManager prints out each failed document entirely, which may cause difficulty in scrolling through the output. It would be possible to fine tune the system output if you catch the exceptions yourself.